src/main/java/org/veupathdb/eda/dumper/io/VariableReader.java

```diff
@@ -0,0 +1,32 @@
+package org.veupathdb.eda.dumper.io;
+
+import org.veupathdb.eda.dumper.model.Variable;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+public class VariableReader<T> implements AutoCloseable {
+  private final InputStream inputStream;
+  private final VariableSerializer<T> variableSerializer;
+
+  public VariableReader(InputStream inputStream, VariableSerializer<T> byteConverter) {
+    this.inputStream = inputStream;
+    this.variableSerializer = byteConverter;
+  }
+
+  /**
+   * Reads a float variable from the wrapped input stream.
+   * @return The next float variable from the input stream.
+   * @throws IOException
+   */
+  public Variable<T> readVar() throws IOException {
+    byte[] bytes = new byte[8];
```

Comment:
I think u might want to define `ENTITY_ID_LENGTH`.  u could use instead of `INT_LENGTH` when we are referring to an entity ID.

here maybe instead of `8` we'd have `ENTITY_ID_LENGTH + FLOAT LENGTH`?